### ISSUE TITLE
Single-Label vs Multi-Label in YOLOv26: Is Dataset Duplication a Mistake?

### ISSUE THREAD

**Issue by sarpx** [ultralytics/ultralytics#24291]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I’m working on a classification setup using a YOLOv26-style pipeline, and I’d appreciate some advice on a dataset strategy change we’re considering.

So far, we’ve been using a **single-label approach**. For each image, we only assign the *dominant* label. For example, if an image is both *“UPSIDE_DOWN”* and *“DARK”*, we only label it as *“UPSIDE_DOWN”* and ignore the other attributes. With this approach, our dataset is around **250k images**.

However, after training, we observed that the model predicts **multiple labels** for about **40k images**, which suggests that the data is inherently **multi-label**, but we forced it into a single-label setup.

Now we are considering changing our approach:

* Allow **multi-label annotations**
* Potentially increase dataset size to ~300k
* Some images may appear **multiple times (2–3x)** with different label combinations

### My concerns:

1. **Data duplication**
   If the same image appears multiple times with different labels, could this lead to overfitting or bias toward those samples?

2. **Better alternative?**
   Instead of duplicating images, would it be more correct to:

   * Keep each image once
   * Use a **multi-hot label vector** (e.g. `[UPSIDE_DOWN=1, DARK=1]`)
   * Train with **sigmoid + BCE loss** instead of softmax?

3. **Practical tradeoff**
   Re-labeling ~40k images for multi-label is costly.
   But duplicating data also feels suboptimal.

4. **Model behavior**
   Since the model is already predicting multiple labels despite single-label training:

   * Is this a strong signal that we should switch to multi-label training?
   * Or can this be handled with threshold tuning alone?

### Summary

* Current: 250k images, single-label (softmax)
* Proposed: multi-label setup, possibly 300k samples with duplicates
* Concern: duplication vs proper multi-label representation

I’d really appreciate insights from anyone who has faced a similar transition, especially in YOLO-based classification setups.

Thanks in advance!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @sarpx, thank you for your interest in Ultralytics 🚀! This is an automated response to help get your issue routed quickly, and an Ultralytics engineer will also assist soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.
If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).
For your classification setup, it would be especially helpful if you could share:
- Your current dataset structure
- A few example labels/images illustrating the single-label vs multi-label case
- The exact training command or script used
- Training/validation metrics and any prediction examples showing the multi-label behavior
- Your current `ultralytics`, Python, and PyTorch versions
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Ultralytics doesn't currently support multi-label classification. You will have to modify Ultralytics to support it. If your classes overlap, then you should ideally use multi-label. Duplicating wouldn't be sufficient.

**Comment by sarpx** [comment]
ok thanks

**Comment by glenn-jocher** [comment]
You're welcome — if you decide to pursue native multi-label classification support, please open a new feature request or [GitHub Discussion](https://github.com/orgs/ultralytics/discussions) with a minimal use case so the team can track it properly.

**Comment by surajsharan** [comment]
This is exactly the use case that PR #24270 addresses: native multi-label 
classification support via a `multi_label=True` flag (sigmoid + BCE), 
currently under review.

To answer your questions directly:

1. **Don't duplicate images** : that will bias training toward those samples. 
   You're right to be concerned.

2. **Multi-hot vector is the correct approach** : one row per image with all 
   labels, sigmoid + BCE loss. This is what #24270 implements.

3. **The model predicting multiple labels despite softmax training is a strong 
   signal** : softmax forces a probability distribution that sums to 1, but 
   your data violates that assumption. Switching to sigmoid + BCE will handle 
   co-occurring labels properly.

4. **Threshold tuning alone won't fix it** :  it just masks the problem. The 
   loss function itself needs to change.

Once #24270 is merged, your setup would be:
```python
model = YOLO("yolo26n-cls.pt")
model.train(data="your_dataset.yaml", multi_label=True, epochs=100)
```

With a `labels.csv` in each split directory mapping image filenames to 
class indices — no duplication needed.

**Comment by glenn-jocher** [comment]
Thanks for linking [PR #24270](https://github.com/ultralytics/ultralytics/pull/24270) — that’s the right place to track native multi-label classification, so since this issue is closed please continue any implementation-specific discussion there.
